### ISSUE TITLE
Compatibility with lsp-ui doc?

### ISSUE THREAD

**Issue by larebsyed** [gonewest818/dimmer.el#49]
LSP UI Doc causes the main buffer to dim which shouldn't happen

**Comment by gonewest818** [comment]
yes, I've got lsp-ui enabled and I see this too. If you have a moment to experiment you could try adding this to your configuration.

    (add-to-list 'dimmer-prevent-dimming-predicates #'lsp-ui-doc--frame-visible-p))

which is exactly what I'll be trying when I get a chance.

**Comment by thermopyle** [comment]
I tried that and it did not quite work, the main buffer is still dimmed when the lsp-ui popup is shown. But sometimes it seems to work, when I move around in the source file and the lsp-ui popup comes and goes sometimes the main buffer is not dimmed.

**Comment by gonewest818** [comment]
Well, it was worth a shot.  I'll have to look in more detail. Fortunately this is one I can reproduce...

**Comment by VirgilSurin** [comment]
I have the same issue and the "hotfix" does not work.
Maybe excluding lsp-ui-doc window might work but I cannot find its name.

**Comment by VirgilSurin** [comment]
UPDATE/HOTFIX
I was messing up with the customization option for lsp-ui and I found out that when you toggle `Lsp Ui Doc Use Childfram` to off, whenever the doc promts, it does not trigger the dimmering of the main active buffer.

**Comment by thermopyle** [comment]
I can confirm this. A bit too early to say without more testing, but this could be an acceptable solution for me. I also think it looks better :-)

**Comment by gonewest818** [comment]
I’ll test as well. Thank you for continuing to look into this!

One thing I can confirm is that child frames are a case I hadn’t considered up to this point.  company-box also uses child frames and has a similar bug.

**Comment by gonewest818** [comment]
Confirmed disabling `lsp-ui-doc-use-childframe` works around the issue for me too.  I still want to solve the underlying issue though, so I'll keep this issue open.

**Comment by buzztaiki** [comment]
Hi, 

In my environment, following snippet has been solved this issue.

```lisp
(defun dimmer-lsp-ui-doc-p ()
  (string-prefix-p " *lsp-ui-doc-" (buffer-name)))
(add-to-list 'dimmer-prevent-dimming-predicates #'dimmer-lsp-ui-doc-p)

(defun advices/dimmer-config-change-handler ()
  (dimmer--dbg-buffers 1 "dimmer-config-change-handler")
  (let ((ignore (cl-some (lambda (f) (and (fboundp f) (funcall f)))
                         dimmer-prevent-dimming-predicates)))
    (dimmer-process-all (not ignore))))
(advice-add 'dimmer-config-change-handler :override #'advices/dimmer-config-change-handler)
```
